infra/sandboxes/computer/control.py

```
@@ -0,0 +1,166 @@
+        try:
+            length = int(self.headers.get("Content-Length", "0"))
+            body = json.loads(self.rfile.read(length))
+            display = body.get("display", ":1")
+            for step in body.get("steps", []):
+                if "waitMs" in step:
+                    time.sleep(max(0, min(int(step["waitMs"]), 5000)) / 1000)
+                    continue
+                source = native_capture(display)
+                handled = source.act(step["argv"]) if source else 0
+                if not handled:
+                    result = subprocess.run(step["argv"], env={**os.environ, "DISPLAY": display})
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)